### PR TITLE
remove session if 401 response

### PR DESCRIPTION
**Pull request by roymoran**
- throw new error with session expired message if 401 status
- replace api request blocks with try catch to read err.message instead of custom message.

### CODE REVIEW

File: src/api/http-client.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
+import { SessionStorageHelper } from '@helpers';
+
 export class HttpClient {
   public static async makeRequest(request: Request): Promise<unknown> {
     const response = await fetch(request);
 
-    // TODO: If 401 - sign out user with
-    // message that session has expired.
+    if (response.status == 401) {
```
Comment (rmjordas) [inline src/api/http-client.ts:7]:
```suggestion
    if (response.status === 401) {
```

File: src/components/projects/project-gallery/project-gallery.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const ProjectGallery: FC<ProjectGalleryProps> = () => {
           setProjects(projectsLookingForMembers);
         } else setError((response.data as ErrorResponse).message);
       } catch (err) {
-        setError('Failed to retrieve a list of projects');
+        setError(err.message);
```
Comment (aaron2212) [inline src/components/projects/project-gallery/project-gallery.tsx:36]: Is it possible that `fetch()` could throw a cryptic error message in cases where it's unable to retrieve the content, and then that is displayed to the user?

Comment (roymoran) [reply src/components/projects/project-gallery/project-gallery.tsx:36]: From my testing the messages weren’t too cryptic - something like “failed to fetch”

But I did rush this implementation so I will test this locally as well as add unit tests to get the checks to pass, thanks for the comment.

Comment (roymoran) [reply src/components/projects/project-gallery/project-gallery.tsx:36]: I’ll also test the response for various status codes e.g. 3xx, 4xx, 500.

**Pull request merged**